app/javascript/components/patient/history/History.js

```diff
@@ -0,0 +1,185 @@
+import { Button, Card, Col, Row } from 'react-bootstrap';
+import ReactTooltip from 'react-tooltip';
+import axios from 'axios';
+
+import DeleteDialog from '../../util/DeleteDialog';
+import reportError from '../../util/ReportError';
+import { formatTimestamp, formatRelativePast } from '../../../utils/DateTime';
+
+class History extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      editMode: false,
+      showDeleteModal: false,
+      comment: props.history.comment,
+    };
+  }
+
+  handleChange = event => {
+    this.setState({ [event.target.id]: event.target.value });
+  };
+
+  toggleEditMode = () => {
+    this.setState({ editMode: true });
+  };
+
+  toggleEditMode = () => {
+    let current = this.state.editMode;
+    this.setState({ editMode: !current, comment: this.props.history.comment });
+  };
+
+  handleEditSubmit = () => {
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .post(window.BASE_PATH + '/histories/' + this.props.history.id + '/edit', {
+          patient_id: this.props.history.patient_id,
+          comment: this.state.comment,
+        })
+        .then(() => {
+          location.reload(true);
+        })
+        .catch(error => {
+          reportError(error);
+        });
+    });
+  };
+
+  toggleDeleteModal = () => {
+    let current = this.state.showDeleteModal;
+    this.setState({
+      showDeleteModal: !current,
+      delete_reason: null,
+      delete_reason_text: null,
+    });
+  };
+
+  handleDeleteSubmit = () => {
+    let deleteReason = this.state.delete_reason;
+    if (deleteReason === 'Other' && this.state.delete_reason_text) {
+      deleteReason += ': ' + this.state.delete_reason_text;
+    }
+
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .post(window.BASE_PATH + '/histories/' + this.props.history.id + '/delete', {
+          patient_id: this.props.history.patient_id,
+          delete_reason: deleteReason,
+        })
+        .then(() => {
+          location.reload(true);
+        })
+        .catch(error => {
+          reportError(error);
+        });
+    });
+  };
+
+  renderEditMode() {
+    return (
+      <React.Fragment>
+        <textarea
+          id="comment"
+          name="comment"
+          className="form-control"
+          aria-label="Edit comment input"
+          style={{ resize: 'none' }}
+          rows="3"
+          value={this.state.comment}
+          onChange={this.handleChange}
+        />
+        <Button
+          id="update-edit-history-btn"
+          variant="primary"
+          size="sm"
+          className="float-right mt-2"
+          disabled={this.state.loading || this.state.comment === '' || this.state.comment === this.props.history.comment}
```

Comment:
I really like this bit of the conditional `this.state.comment === this.props.history.comment`. 👍 👍